### PR TITLE
feat: [PLG-2860]: integrating canny sdk for authetication

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by kapil-harness — DISMISSED** [review]

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -37,6 +38,7 @@ sed -i "s|<\!-- newNavContentfulEnvironment -->|<script>window.newNavContentfulE
 sed -i "s|<\!-- harnessNameSpacePlaceHolder -->|<script>window.harnessNameSpace = '$HARNESS_NAME_SPACE'</script>|" index.html
 sed -i "s|<\!-- harnessClusterURLPlaceHolder -->|<script>window.harnessClusterURL = '$HARNESS_CLUSTER_URL'</script>|" index.html
 sed -i "s|<\!-- stripeApiKey -->|<script>window.stripeApiKey = '$STRIPE_API_KEY'</script>|" index.html
+sed -i "s|<\!-- harnessAppId -->|<script>window.harnessAppId = '$HARNESS_APP_ID'</script>|" index.html
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:41]: for us this should be `CANNY_APP_ID`

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -54,7 +56,7 @@ fi
...
+  sed -i "s|#cspHeadersPlaceholder|add_header Content-Security-Policy-Report-Only \"script-src 'self' 'unsafe-inline' https://canny.io/sdk.js https://*.harness.io https://widget.saberfeedback.com https://cdn.segment.com https://js.refiner.io https://widget.intercom.io https://js.intercomcdn.com https://d2wy8f7a9ursnm.cloudfront.net https://cdn.appdynamics.com; style-src 'self' 'unsafe-inline' https://*.harness.io https://fonts.googleapis.com https://widget.saberfeedback.com https://js.refiner.io; img-src 'self' data: blob: https://*.harness.io https://widget.saberfeedback.com; font-src 'self' https://fonts.gstatic.com;\";|" $NGINX_CONFIG_FILE
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:59]: you only need to add the origin.. adding the filename will not work.

File: src/modules/10-common/components/ResourceCenter/ResourceCenter.tsx [hunk 3]
```diff
@@ -56,6 +58,9 @@ export const ResourceCenter: React.FC<ResourceCenterProps> = ({ link, isOpen: di
   const [buttonDisabled, setButtonDisabled] = useState(false)
   const [show, setShow] = useState<boolean>(Boolean(display))
   const { showModal } = useReleaseNotesModal()
+  const { pathname } = useLocation()
+  const { accounts } = currentUserInfo
+  const account = accounts?.find(({ uuid }) => pathname.includes(uuid as string)) || {}
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/ResourceCenter.tsx:63]: this is not needed.. `accountInfo` is already available in AppStoreContext.

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@
  */
 
 import moment from 'moment'
-import type { UserInfo } from 'services/cd-ng'
+import type { GatewayAccountRequestDTO, UserInfo } from 'services/cd-ng'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:9]: we should use `AccountDTO`, the same that's used in AppStoreContext

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

**Pull request closed without merging**